src/SpecFlow.NetCore/Args.cs

```diff
@@ -1,79 +1,91 @@
 ﻿using System;
+using System.Collections.Generic;
 using System.IO;
 using System.Linq;
 using static System.Console;
-
+ 
 namespace Specflow.NetCore
 {
 	internal class Args
 	{
-		public const string SpecFlowPathArgName = "--specflow-path";
-		public const string WorkingDirectoryArgName = "--working-directory";
-		public const string TestFrameworkArgName = "--test-framework";
-		public const string ToolsVersionArgName = "--tools-version";
-
-		public string SpecFlowPath { get; }
-		public DirectoryInfo WorkingDirectory { get; }
-		public string TestFramework { get; }
-		public string ToolsVersion { get; }
-
-		public Args(string[] args)
-		{
-			WorkingDirectory = new DirectoryInfo(Directory.GetCurrentDirectory());
-
-			if (args == null || !args.Any())
-				return;
-
-			// Very basic arg parsing:
-			//   - Assume odd elements are arg names.
-			//   - Assume even elements are arg values.
-			//   - Paths with spaces will probably blow up.
-			//   - Duplicates, last one wins.
-
-			for (var i = 0; i < args.Length; i++)
+ 		public const string SpecFlowPathArgName = "--specflow-path";
+ 		public const string WorkingDirectoryArgName = "--working-directory";
+ 		public const string TestFrameworkArgName = "--test-framework";
+ 		public const string ToolsVersionArgName = "--tools-version";
+ 
+ 		public string SpecFlowPath { get; }
+ 		public DirectoryInfo WorkingDirectory { get; }
+ 		public string TestFramework { get; }
+ 		public string ToolsVersion { get; }
+ 
+ 		public Args(string[] args)
+ 		{
+ 			WorkingDirectory = new DirectoryInfo(Directory.GetCurrentDirectory());
+ 
+ 			// Command line arguments are required.  If absent, exit.
+ 			if (args == null || !args.Any())
+ 			{
+ 				return;
+ 			}
+ 
+ 			// establish a dictionary of all good command line variables
+ 			var argDictionary = new Dictionary<string, string>()
 			{
-				if (IsOdd(i))
-					continue;
-
-				if (i + 1 >= args.Length)
-					throw new Exception("Uneven arguments");
-
-				var name = args[i];
-				var value = args[i + 1];
-
-				switch (name)
+ 				{SpecFlowPathArgName , null},
+ 				{WorkingDirectoryArgName, null },
+ 				{TestFrameworkArgName, null },
+ 				{ToolsVersionArgName, null }
+ 			};
+ 
+ 
+ 			string lastKey = null;
+ 			// loop through all arguments, attempting to detect and fix bad values
+ 			for (var x = 0; x<args.Length; x++)

```

Comment:
Nit: It's idiomatic to use ` i` for variable name.

Reply:
Updated to be a foreach.